freestyle-effects/shared/src/main/scala/effects/traverse.scala

```diff
@@ -0,0 +1,44 @@
+package io.freestyle.effects
+
+import io.freestyle._
+import cats.{~>, MonadCombine, Traverse}
+import cats.data.ValidatedNel
+import cats.free.Free
+
+import cats.implicits._
+
+object traverse {
+
+  final class TraverseProvider[G[_]] {
+
+    /** Acts as a generator providing traversable semantics to programs
+     */
+    @free sealed abstract class TraverseM[F[_]] {
+      def empty[A]: FreeS[F, A]
+      def singleton[A](a: A): FreeS[F, A]
+      def fromTraversable[A](ta: G[A]): FreeS[F, A]
+    }
+
+    /** Interpretable as long as traverse instance for G[_] and a monad combine for G[_] exists
+     * in scope
+     */
+    object implicits {
+      implicit def freestyleTraverseMInterpreter[F[_], M[_]](
+          implicit MC: MonadCombine[M],
+          FT: Traverse[G]): TraverseM.Interpreter[M] =
+        new TraverseM.Interpreter[M] {
+          def emptyImpl[A]: M[A]           = MC.empty[A]
+          def singletonImpl[A](a: A): M[A] = MC.pure[A](a)
+          def fromTraversableImpl[A](ta: G[A]): M[A] = {
```

Comment:
I think that this should be more efficient than the combination of `traverse` and `unite` :

```scala
val monoidMA = MonoidK[M].algebra[A] 
Foldable[G].foldMap(ta)(Applicative[M].pure)(monoidMA)

// or shorter
val monoidMA = MC.algebra[A] 
ta.foldMap(MC.pure)(monoidMA)
```

Which also means that you only need `Foldable[G]` (instead of `Traverse[G]`), I think ?